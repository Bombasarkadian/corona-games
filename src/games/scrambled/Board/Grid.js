import React from "react";
import PropTypes from "prop-types";

import Field from "./Field";

const propTypes = {
  board: PropTypes.array.isRequired,
  playerLetters: PropTypes.array.isRequired,
  clickable: PropTypes.func,
  handleFieldClick: PropTypes.func,
  selectionEnabled: PropTypes.bool,
};
const Grid = ({ board, playerLetters, clickable, handleFieldClick, selectionEnabled }) => {
  const size = 46;
  const bordersColor = selectionEnabled ? "#b1b1b1" : "#E4E4E4";

  return (
    <div
      style={{
        background: "#717171",
        userSelect: "none",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        width: 18.5 * size,
        height: 18.5 * size,
        borderRadius: 3,
      }}
    >
      <table
        style={{
          border: `2px solid ${bordersColor}`,
          borderRadius: 5,
          background: bordersColor,
          boxShadow: "rgba(0, 0, 0, 0.4) 0px 0px 10px 2px",
          borderSpacing: 5,
          transition: "background 0.3s 0.08s, border-color 0.3s 0.08s",
        }}
      >
        <tbody>
          {board.map((rowContainer, y) => {
            return (
              <tr key={y} style={{ height: size }}>
                {rowContainer.row.map((el, x) => {
                  return (
                    <td
                      key={`${x},${y}`}
                      style={{
                        width: size,
                        height: size,
                        padding: 0,
                        position: "relative",
                      }}
                    >
                      <Field
                        base={el}
                        overlay={
                          playerLetters &&
                          playerLetters.find((tile) => tile.x === x && tile.y === y)
                        }
                        handleFieldClick={() => handleFieldClick(x, y)}
                        clickable={clickable(x, y)}
                        selectionEnabled={selectionEnabled}
                      />
                    </td>
                  );
                })}
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
};
Grid.propTypes = propTypes;

export default Grid;
