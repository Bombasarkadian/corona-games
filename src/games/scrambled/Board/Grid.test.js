import React from "react";
import renderer from "react-test-renderer";
import Grid from "./Grid";

describe("scrambled Grid component", () => {
  it("renders correctly when empty", () => {
    expect.hasAssertions();
    const component = renderer.create(
      <Grid
        board={[{ row: [0, 0, 0] }, { row: [0, 0, 0] }, { row: [0, 0, 0] }]}
        playerTiles={[]}
        clickable={jest.fn()}
        handleFieldClick={jest.fn()}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with some tiles", () => {
    expect.hasAssertions();
    const component = renderer.create(
      <Grid
        board={[
          {
            row: [
              { letter: "A", points: 1 },
              0,
              { bonus: { type: "letter", multiply: 3 }, start: true },
            ],
          },
          { row: [{ bonus: { type: "word", multiply: 2 } }, 0, 0] },
          { row: [0, 0, 0] },
        ]}
        playerTiles={[]}
        clickable={jest.fn()}
        handleFieldClick={jest.fn()}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with temporary tiles", () => {
    expect.hasAssertions();
    const component = renderer.create(
      <Grid
        board={[
          {
            row: [
              { letter: "A", points: 1 },
              0,
              { bonus: { type: "letter", multiply: 3 }, start: true },
            ],
          },
          { row: [{ bonus: { type: "word", multiply: 2 } }, 0, 0] },
          { row: [0, 0, 0] },
        ]}
        playerTiles={[
          { letter: "B", points: 3, x: 1, y: 0 },
          { letter: null, points: 0, replacement: "C", x: 2, y: 0 },
        ]}
        clickable={jest.fn()}
        handleFieldClick={jest.fn()}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with preview tiles", () => {
    expect.hasAssertions();
    const component = renderer.create(
      <Grid
        board={[
          {
            row: [
              { letter: "A", points: 1 },
              0,
              { bonus: { type: "letter", multiply: 3 }, start: true },
            ],
          },
          { row: [{ bonus: { type: "word", multiply: 2 } }, 0, 0] },
          { row: [0, 0, 0] },
        ]}
        playerTiles={[
          { x: 1, y: 0, preview: true },
          { x: 2, y: 0, preview: true },
        ]}
        clickable={jest.fn()}
        handleFieldClick={jest.fn()}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
