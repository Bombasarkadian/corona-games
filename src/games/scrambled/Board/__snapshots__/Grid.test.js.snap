// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrambled Grid component renders correctly when empty 1`] = `
<div
  class="grid"
>
  <table>
    <tbody>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`scrambled Grid component renders correctly with preview tiles 1`] = `
<div
  class="grid"
>
  <table>
    <tbody>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="tile"
            >
              <div
                class="letter"
              >
                A
              </div>
              <div
                class="points"
              >
                1
              </div>
            </div>
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="tile tile_highlighted"
            >
              <div
                class="letter"
              >
                ?
              </div>
              <div
                class="points"
              />
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
            style="background: rgb(22, 145, 159);"
          >
            <div
              class="tile tile_highlighted"
            >
              <div
                class="letter"
              >
                ?
              </div>
              <div
                class="bonus"
                style="background: rgb(22, 145, 159);"
              />
              <div
                class="points"
              />
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
            style="background: rgb(249, 197, 74);"
          >
            <div
              class="bonus"
            >
              <div
                class="bonus_multiply"
              >
                2
                x
              </div>
              <div
                class="bonus_type"
              >
                game.info.bonuses.word
              </div>
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`scrambled Grid component renders correctly with some tiles 1`] = `
<div
  class="grid"
>
  <table>
    <tbody>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="tile"
            >
              <div
                class="letter"
              >
                A
              </div>
              <div
                class="points"
              >
                1
              </div>
            </div>
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
            style="background: rgb(22, 145, 159);"
          >
            <i
              aria-hidden="true"
              class="star big disabled fitted icon"
            />
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
            style="background: rgb(249, 197, 74);"
          >
            <div
              class="bonus"
            >
              <div
                class="bonus_multiply"
              >
                2
                x
              </div>
              <div
                class="bonus_type"
              >
                game.info.bonuses.word
              </div>
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`scrambled Grid component renders correctly with temporary tiles 1`] = `
<div
  class="grid"
>
  <table>
    <tbody>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="tile"
            >
              <div
                class="letter"
              >
                A
              </div>
              <div
                class="points"
              >
                1
              </div>
            </div>
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="tile tile_highlighted"
            >
              <div
                class="letter"
              >
                B
              </div>
              <div
                class="points"
              >
                3
              </div>
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
            style="background: rgb(22, 145, 159);"
          >
            <div
              class="tile tile_highlighted"
            >
              <div
                class="letter letter_replacement"
              >
                C
              </div>
              <div
                class="bonus"
                style="background: rgb(22, 145, 159);"
              />
              <div
                class="points"
                style="text-shadow: -1px -1px 0 #16919F, 1px -1px 0 #16919F, -1px  1px 0 #16919F, 1px  1px 0 #16919F;"
              >
                0
              </div>
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
            style="background: rgb(249, 197, 74);"
          >
            <div
              class="bonus"
            >
              <div
                class="bonus_multiply"
              >
                2
                x
              </div>
              <div
                class="bonus_type"
              >
                game.info.bonuses.word
              </div>
            </div>
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
      <tr
        style="height: 46px;"
      >
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
        <td
          class="cell"
        >
          <div
            class="field"
          >
            <div
              class="field dimmer"
            />
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
